clash-cores/src/Clash/Cores/Xilinx/DcFifo.hs

```diff
@@ -0,0 +1,6 @@
+module Clash.Cores.Xilinx.DcFifo
+  ( module Clash.Cores.Xilinx.DcFifo.Implementation
+  ) where
```

Comment:
Could you please explain the purpose of putting everything in `DcFifo.Implementation` instead of in `DcFifo`? This makes the Haddock more annoying to use, IMO. Suppose a user wishes to look up documentation for DcFifo. They might open the Haddock index page for the `clash-cores` package, then scroll through the module list and click `Clash.Cores.Xilinx.DcFifo`. They are greeted with this screen:

![image](https://user-images.githubusercontent.com/3692925/181017653-84fa3a3a-373c-4281-b2ed-b7deff023dd6.png)

The sole purpose of this screen is to say to the user "you need to click this link to go where you probably wanted to go in the first place".

Now, we could do some tricks so the actual function documentation is on that page anyway. But I wonder why it isn't there to begin with? In what way is it good for our users to have all the functions in `DcFifo.Implementation`, or what relevance does it have for us developers?

Reply:
This was so that we could use TemplateHaskell to generate instances in `DcFifo.Implementation`.

I replaced the template Haskell with its splice; I think the result is much nicer now.